cli/tests/compose.bats

```diff
@@ -0,0 +1,82 @@
+
+#! /usr/bin/env bats
+# -*- mode: bats; -*-
+# ============================================================================ #
+#
+# Test environment composition
+#
+#
+# ---------------------------------------------------------------------------- #
+
+load test_support.bash
+
+# bats file_tags=compose
+
+# ---------------------------------------------------------------------------- #
+
+setup_file() {
+  common_file_setup
+}
+
+setup() {
+  common_test_setup
+  home_setup test # Isolate $HOME for each test.
+
+  # concurrent pkgdb database creation
+  # todo: still needed now that there is no pkgdb?
+  setup_isolated_flox
```

Comment:
I seem to recall that we're mostly depending on this for `wait_for_watchdogs` now and we're calling that in `teardown` so we should drop this part of the comment.
```suggestion
  setup_isolated_flox
```